### PR TITLE
[Feature] Support padding for logits and unequal batch size for logits and bitmask

### PR DESCRIPTION
**Pull request by Ubospica**
This PR supports:
1. Padding on the vocabulary dimension for logits. vLLM could introduce such padding and this is not supported by the previous kernel.
2. Unequal batch size for logits and bitmask when indices are specified. When indices are not specified, we require the batch sizes for logits and bitmask the same. When indices are specified, we only require the indices larger than

### CODE REVIEW

**Comment by Ubospica** [comment]
Old performance:

New performance:
```
Batch:  1 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 58.8940
Batch:  1 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 598.7030
Batch:  1 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 594.7130
Batch:  1 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 543.3010
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 984.1650
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 33464.3500
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 4 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 344.0990
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 4 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 8495.6430
Batch: 64 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 33614.0180
Batch: 64 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cpu    | Execution time (μs): 33769.6170
Batch:  1 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 6.6918
Batch:  1 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 6.8836
Batch:  1 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 7.0885
Batch:  1 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 7.0889
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 14.3056
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 27.4631
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 4 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 94.8330
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 4 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 95.4378
Batch: 64 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 37.6485
Batch: 64 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: cuda   | Execution time (μs): 37.6297
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float16   | Impl: cuda   | Execution time (μs): 12.3645
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.bfloat16  | Impl: cuda   | Execution time (μs): 12.3742
Batch:  1 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 6.3034
Batch:  1 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 7.0350
Batch:  1 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 6.4437
Batch:  1 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 6.4452
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 14.9647
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 42.8901
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 4 | DType: torch.float32   | Impl: triton | Execution time (μs): 191.2202
Batch: 64 | Vocab: 128000 | Masked: 120000 | Stride: 4 | DType: torch.float32   | Impl: triton | Execution time (μs): 183.8002
Batch: 64 | Vocab: 128001 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 34.3136
Batch: 64 | Vocab: 128010 | Masked: 120000 | Stride: 1 | DType: torch.float32   | Impl: triton | Execution time (μs): 34.2011
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.float16   | Impl: triton | Execution time (μs): 14.5390
Batch: 64 | Vocab: 128000 | Masked:   1024 | Stride: 1 | DType: torch.bfloat16  | Impl: triton | Execution time (μs): 14.5105
```
